### PR TITLE
Fix: subdirectory deployment

### LINKED ISSUE #2128: [Bug] base and startUrl invalid
### Description

base and startUrl invalid setting 

### Steps to reproduce

After setting the base: http://host.local/homepage
After that, we can only pass through http://host.local/ Unable to access homepage

### homepage version

v 0.7.2

### Installation method

Docker

### Configuration

```yaml
base: http://host.local/homepage
startUrl: http://host.local/homepage
```

### Container Logs

_No response_

### Browser Logs

_No response_

### Troubleshooting

After setting the base: http://host.local/homepage
After that, we can only pass through http://host.local/ Unable to access homepage

### Other

_No response_

### Before submitting, I have made sure to

- [X] Check [the documentation](https://gethomepage.dev/)
- [X] Follow [the troubleshooting guide](https://gethomepage.dev/latest/more/troubleshooting/) (please include output above if applicable).
- [X] Search [existing issues](https://github.com/gethomepage/homepage/search?q=&type=issues) and [discussions](https://github.com/gethomepage/homepage/search?q=&type=discussions).

### PR DESCRIPTION
**Pull request by alfred-nice**
## Proposed change

Closes #2128 #539

## Type of change

- [ ] New service widget
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [ ] If adding a service widget or a change that requires it, I have added corresponding documentation changes.
- [ ] If adding a new widget I have reviewed the [guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [ ] If applicable, I have checked that all tests pass with e.g. `pnpm lint`.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

tips:
next.config.js. add `assetPrefix: ".",`
others. mod `/api/xxx` to `api/xxx`.
thanks for @benphelps @shamoon guides. #2194

I am learning to contribute code & PR. If there are any mistakes, I kindly request guidance from everyone.

### CODE REVIEW

**Comment by shamoon** [comment]
I have no idea why https://github.com/gethomepage/homepage/pull/2194 was merged, its an empty PR and was blocked, huh?

**Comment by shamoon** [comment]
Anyway, thanks for re-creating

File: next.config.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const { i18n } = require("./next-i18next.config");
 
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  assetPrefix: ".",
```
Comment (shamoon) [inline next.config.js:5]: This is the only thing I'm unsure of, where does this change come from / why is it needed?

Comment (alfred-nice) [reply next.config.js:5]: > This is the only thing I'm unsure of, where does this change come from / why is it needed?

Subdirectory deployment need the realtive path. like _next/static, not /_next/static path.

for example:
http://xxxx/_next/static/chunks/src_components_toggles_theme_jsx.js
http://xxxxx/homepage/_next/static/chunks/src_components_toggles_theme_jsx.js

Comment (shamoon) [reply next.config.js:5]: Oh i see, just chucks it into the path in the asset `src`

**Review by shamoon — APPROVED** [review]
In general this seems fine to me. I havent tried setting up a subdirectory deployment however, because I find them to be a bit of a pain.

In general seems safe, so OK by me, but curious if @benphelps has any other thoughts

**Comment by shamoon** [comment]
you might want to disable the pull bot on your fork, seems to break your PRs. Or you also might avoid this by making PRs from a branch not main, but I’m not sure

**Comment by benphelps** [comment]
> In general this seems fine to me. I havent tried setting up a subdirectory deployment however, because I find them to be a bit of a pain.
> 
> In general seems safe, so OK by me, but curious if @benphelps has any other thoughts

I'm sure this will have some adverse effects for a handful of users, but is worth the trouble to get this recurring issue taken care of.  Looks fine.

**Comment by alfred-nice** [comment]
> you might want to disable the pull bot on your fork, seems to break your PRs. Or you also might avoid this by making PRs from a branch not main, but I’m not sure

sorry. i see. I will disable the pull bot.

**Comment by shamoon** [comment]
@alfred-nice unfortunately we're going to revert this change for now, it does seem to break the font and the fix is not obvious, sort of a catch-22 with importing the font files into the css and a relative URL.

Of course if you can solve it we'd be happy to merge a subsequent PR.

Thanks for your effort.

**Comment by alfchao** [comment]
> @alfred-nice unfortunately we're going to revert this change for now, it does seem to break the font and the fix is not obvious, sort of a catch-22 with importing the font files into the css and a relative URL.
> 
> Of course if you can solve it we'd be happy to merge a subsequent PR.
> 
> Thanks for your effort.

When subdirectory deployment, the correct path should be **/homepage/_next/static/media/Manrope.63012343.woff2** (200 good), not **/homepage/_next/static/css/_next/static/media/Manrope.63012343.woff2** (404 bad).

I'm not sure where this font was imported from, and I don't have much knowledge about Next.js. I would greatly appreciate it if someone could provide me with guidance on this matter.

Thanks!

**Comment by waltlawsmacd** [comment]
@alfchao I believe the font is imported in [this file](https://github.com/gethomepage/homepage/blob/ebd384c62dcff0e0ec4b27097f4b9baee83a0bdb/src/styles/manrope.css) - I am also don't know much about Next.js, but I'm guessing the relative path is what's causing the broken link

**Pull request merged**